### PR TITLE
fixes and housekeeping

### PR DESCRIPTION
**Pull request by 17Halbe**
moved the fail2ban function into its own script into the container
moved the creation of the fail-auth-mailer container into the Makefile(where it belongs! ;)

### CODE REVIEW

**Comment by johansmitsnl** [comment]
Could you verify the tests, travis failed twice, or do they run good locally?

**Comment by 17Halbe** [comment]
checks completed twice without a problem.. running them again now. Who knows, what I really tested!? ;)

File: test/tests.bats [hunk 1]
```diff
@@ -1194,16 +1191,16 @@ load 'test_helper/bats-assert/load'
   assert_success
 }
 @test "checking setup.sh: setup.sh debug fail2ban" {
-  
+
   run docker exec mail_fail2ban /bin/sh -c "fail2ban-client set dovecot banip 192.0.66.4"
   run docker exec mail_fail2ban /bin/sh -c "fail2ban-client set dovecot banip 192.0.66.5"
-  sleep 10
+  sleep 2
```
Comment (17Halbe) [inline test/tests.bats]: Is this change the Problem? Although the error with Travis happens on a very different spot..

**Comment by johansmitsnl** [comment]
@17Halbe could you resolve the conflict?

**Comment by 17Halbe** [comment]
Nope, all tests have been working fine on my machine, I ran them liek 4 times now.. It's not an important upgrade anyway. I might pull it and submit several granular ones. :+1: ?

**Pull request closed without merging**